Exception for DOI search in case DOI contains masked characters

### JabRef version

5.6 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

Similar issue to #8786 in case of a DOI search (masked underscore) in DOI.

### Appendix

...
<details>

<summary>Log File</summary>

```
org.jabref.logic.importer.FetcherException: No DOI data exists
	at org.jabref@5.6.60000/org.jabref.logic.importer.fetcher.DoiFetcher.performSearchById(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.gui.mergeentries.FetchAndMergeEntry.lambda$fetchAndMerge$0(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.gui.util.BackgroundTask$1.call(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.gui.util.DefaultTaskExecutor$1.call(Unknown Source)
	at org.jabref.merged.module@5.6.60000/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.logic.net.URLDownload.asString(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.logic.net.URLDownload.asString(Unknown Source)
	... 11 more
Caused by: java.io.FileNotFoundException: https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/java.net.HttpURLConnection.getResponseCode(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source)
	at org.jabref@5.6.60000/org.jabref.logic.net.URLDownload.openConnection(Unknown Source)
	... 13 more
```

</details>

You are right. The error is different. Excuse my trigger happy fingers pushing buttons too fast.

`Caused by: java.io.FileNotFoundException: https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3`

--- a/.github/workflows/automerge.yml
+++ b/.github/workflows/automerge.yml
@@ -21,7 +21,7 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Merge pull requests
-        uses: pascalgn/automerge-action@v0.15.3
+        uses: pascalgn/automerge-action@v0.15.2
         if: steps.waitforstatuschecks.outputs.status == 'success'
         env:
           MERGE_METHOD: "merge"

--- a/.github/workflows/automerge.yml
+++ b/.github/workflows/automerge.yml
@@ -21,7 +21,7 @@ jobs:
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Merge pull requests
-        uses: pascalgn/automerge-action@v0.15.2
+        uses: pascalgn/automerge-action@v0.15.3
         if: steps.waitforstatuschecks.outputs.status == 'success'
         env:
           MERGE_METHOD: "merge"

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -29,7 +29,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed the unnecessary horizontal scroll bar in group panel [#8467](https://github.com/JabRef/jabref/issues/8467)
 - We fixed an issue where the notification bar message, icon and actions appeared to be invisible. [#8761](https://github.com/JabRef/jabref/issues/8761)
 - We fixed an issue where deprecated fields tab is shown when the fields don't contain any values. [#8396](https://github.com/JabRef/jabref/issues/8396)
-- We fixed an issue that there are exceptions for DOI search in case DOI contains masked characters. [#8787](https://github.com/JabRef/jabref/issues/8787)
+- We fixed an issue where an exxception for DOI search occured when the DOI contained urlencoded characters. [#8787](https://github.com/JabRef/jabref/issues/8787)
 - We fixed an issue which allow us to select and open identifiers from a popup list in the maintable [#8758](https://github.com/JabRef/jabref/issues/8758), [8802](https://github.com/JabRef/jabref/issues/8802)
 - We fixed an issue where the escape button had no functionality within the "Filter groups" textfield. [koppor#562](https://github.com/koppor/jabref/issues/562)
 - We fixed an issue where right clicking a group and choose "remove selected entries from this group" leads to error when Bibtex source tab is selected. [#8012](https://github.com/JabRef/jabref/issues/8012)

--- a/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
+++ b/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
@@ -78,11 +78,4 @@ public class DoiCleanup implements CleanupJob {
         CleanupJob eraser = new FieldFormatterCleanup(field, new ClearFormatter());
         changes.addAll(eraser.cleanup(entry));
     }
-
-    private String decodeDoi(String doiValue) throws UnsupportedEncodingException {
-        if (doiValue == null) {
-            return null;
-        }
-        return URLDecoder.decode(doiValue, "UTF-8");
-    }
 }

--- a/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
+++ b/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
@@ -27,7 +27,6 @@ public class DoiCleanup implements CleanupJob {
 
     @Override
     public List<FieldChange> cleanup(BibEntry entry) {
-
         List<FieldChange> changes = new ArrayList<>();
 
         // First check if the Doi Field is empty

--- a/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
+++ b/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
@@ -36,8 +36,7 @@ public class DoiCleanup implements CleanupJob {
             String decodeDoiFieldValue = "";
             try {
                 decodeDoiFieldValue = URLDecoder.decode(doiFieldValue, "UTF-8");
-            } catch (
-                    UnsupportedEncodingException e) {
+            } catch (UnsupportedEncodingException e) {
                 decodeDoiFieldValue = doiFieldValue;
             }
             doiFieldValue = decodeDoiFieldValue;

--- a/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
+++ b/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
@@ -16,36 +16,12 @@ import org.jabref.model.entry.field.UnknownField;
 import org.jabref.model.entry.identifier.DOI;
 
 /**
- * Formats
- * the
- * DOI
- * (e.g.
- * removes
- * http
- * part)
- * and
- * also
- * moves
- * DOIs
- * from
- * note,
- * url
- * or
- * ee
- * field
- * to
- * the
- * doi
- * field.
+ * Formats the DOI (e.g. removes http part) and also moves DOIs from note, url or ee field to the doi field.
  */
 public class DoiCleanup implements CleanupJob {
 
     /**
-     * Fields
-     * to
-     * check
-     * for
-     * DOIs.
+     * Fields to check for DOIs.
      */
     private static final List<Field> FIELDS = Arrays.asList(StandardField.NOTE, StandardField.URL, new UnknownField("ee"));
 

--- a/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
+++ b/src/main/java/org/jabref/logic/cleanup/DoiCleanup.java
@@ -1,5 +1,7 @@
 package org.jabref.logic.cleanup;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
@@ -14,22 +16,56 @@ import org.jabref.model.entry.field.UnknownField;
 import org.jabref.model.entry.identifier.DOI;
 
 /**
- * Formats the DOI (e.g. removes http part) and also moves DOIs from note, url or ee field to the doi field.
+ * Formats
+ * the
+ * DOI
+ * (e.g.
+ * removes
+ * http
+ * part)
+ * and
+ * also
+ * moves
+ * DOIs
+ * from
+ * note,
+ * url
+ * or
+ * ee
+ * field
+ * to
+ * the
+ * doi
+ * field.
  */
 public class DoiCleanup implements CleanupJob {
 
     /**
-     * Fields to check for DOIs.
+     * Fields
+     * to
+     * check
+     * for
+     * DOIs.
      */
     private static final List<Field> FIELDS = Arrays.asList(StandardField.NOTE, StandardField.URL, new UnknownField("ee"));
 
     @Override
     public List<FieldChange> cleanup(BibEntry entry) {
+
         List<FieldChange> changes = new ArrayList<>();
 
         // First check if the Doi Field is empty
         if (entry.hasField(StandardField.DOI)) {
             String doiFieldValue = entry.getField(StandardField.DOI).orElse(null);
+
+            String decodeDoiFieldValue = "";
+            try {
+                decodeDoiFieldValue = URLDecoder.decode(doiFieldValue, "UTF-8");
+            } catch (
+                    UnsupportedEncodingException e) {
+                decodeDoiFieldValue = doiFieldValue;
+            }
+            doiFieldValue = decodeDoiFieldValue;
 
             Optional<DOI> doi = DOI.parse(doiFieldValue);
 
@@ -45,7 +81,7 @@ public class DoiCleanup implements CleanupJob {
                 // Doi field seems to contain Doi -> cleanup note, url, ee field
                 for (Field field : FIELDS) {
                     entry.getField(field).flatMap(DOI::parse)
-                        .ifPresent(unused -> removeFieldValue(entry, field, changes));
+                         .ifPresent(unused -> removeFieldValue(entry, field, changes));
                 }
             }
         } else {
@@ -67,5 +103,12 @@ public class DoiCleanup implements CleanupJob {
     private void removeFieldValue(BibEntry entry, Field field, List<FieldChange> changes) {
         CleanupJob eraser = new FieldFormatterCleanup(field, new ClearFormatter());
         changes.addAll(eraser.cleanup(entry));
+    }
+
+    private String decodeDoi(String doiValue) throws UnsupportedEncodingException {
+        if (doiValue == null) {
+            return null;
+        }
+        return URLDecoder.decode(doiValue, "UTF-8");
     }
 }

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -18,19 +18,7 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Class
- * for
- * working
- * with
- * <a
- * href="https://en.wikipedia.org/wiki/Digital_object_identifier">Digital
- * object
- * identifiers
- * (DOIs)</a>
- * and
- * <a
- * href="http://shortdoi.org">Short
- * DOIs</a>
+ * Class for working with <a href="https://en.wikipedia.org/wiki/Digital_object_identifier">Digital object identifiers (DOIs)</a> and <a href="http://shortdoi.org">Short DOIs</a>
  */
 public class DOI implements Identifier {
 
@@ -117,38 +105,10 @@ public class DOI implements Identifier {
     private boolean isShortDoi = false;
 
     /**
-     * Creates
-     * a
-     * DOI
-     * from
-     * various
-     * schemes
-     * including
-     * URL,
-     * URN,
-     * and
-     * plain
-     * DOIs/Short
-     * DOIs.
-     *
-     * @param doi the
-     *            DOI/Short
-     *            DOI
-     *            string
-     * @throws NullPointerException     if
-     *                                  DOI/Short
-     *                                  DOI
-     *                                  is
-     *                                  null
-     * @throws IllegalArgumentException if
-     *                                  doi
-     *                                  does
-     *                                  not
-     *                                  include
-     *                                  a
-     *                                  valid
-     *                                  DOI/Short
-     *                                  DOI
+     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs.
+     * @param doi the DOI/Short DOI string
+     * @throws NullPointerException     if DOI/Short DOI is null
+     * @throws IllegalArgumentException if doi does not include a valid DOI/Short DOI
      */
     public DOI(String doi) {
         Objects.requireNonNull(doi);
@@ -193,46 +153,10 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Creates
-     * an
-     * Optional&lt;DOI>
-     * from
-     * various
-     * schemes
-     * including
-     * URL,
-     * URN,
-     * and
-     * plain
-     * DOIs.
-     * <p>
-     * Useful
-     * for
-     * suppressing
-     * the
-     * <c>IllegalArgumentException</c>
-     * of
-     * the
-     * Constructor
-     * and
-     * checking
-     * for
-     * Optional.isPresent()
-     * instead.
-     *
-     * @param doi the
-     *            DOI/Short
-     *            DOI
-     *            string
-     * @return an
-     * Optional
-     * containing
-     * the
-     * DOI
-     * or
-     * an
-     * empty
-     * Optional
+     * Creates an Optional&lt;DOI> from various schemes including URL, URN, and plain DOIs.
+     * Useful for suppressing the <c>IllegalArgumentException</c> of the Constructor and checking for Optional.isPresent() instead.
+     * @param doi the DOI/Short DOI string
+     * @return an Optional containing the DOI or an empty Optional
      */
     public static Optional<DOI> parse(String doi) {
         try {
@@ -251,61 +175,18 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Determines
-     * whether
-     * a
-     * DOI/Short
-     * DOI
-     * is
-     * valid
-     * or
-     * not
-     *
-     * @param doi the
-     *            DOI/Short
-     *            DOI
-     *            string
-     * @return true
-     * if
-     * DOI
-     * is
-     * valid,
-     * false
-     * otherwise
+     * Determines whether a DOI/Short DOI is valid or not
+     * @param doi the DOI/Short DOI string
+     * @return true if DOI is valid, false otherwise
      */
     public static boolean isValid(String doi) {
         return parse(doi).isPresent();
     }
 
     /**
-     * Tries
-     * to
-     * find
-     * a
-     * DOI/Short
-     * DOI
-     * inside
-     * the
-     * given
-     * text.
-     *
-     * @param text the
-     *             Text
-     *             which
-     *             might
-     *             contain
-     *             a
-     *             DOI/Short
-     *             DOI
-     * @return an
-     * Optional
-     * containing
-     * the
-     * DOI
-     * or
-     * an
-     * empty
-     * Optional
+     * Tries to find a DOI/Short DOI inside the given text.
+     * @param text the Text which might contain a DOI/Short DOI
+     * @return an Optional containing the DOI or an empty Optional
      */
     public static Optional<DOI> findInText(String text) {
         Optional<DOI> result = Optional.empty();
@@ -337,63 +218,24 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Return
-     * the
-     * plain
-     * DOI/Short
-     * DOI
-     *
-     * @return the
-     * plain
-     * DOI/Short
-     * DOI
-     * value.
+     * Return the plain DOI/Short DOI
+     * @return the plain DOI/Short DOI value.
      */
     public String getDOI() {
         return doi;
     }
 
     /**
-     * Determines
-     * whether
-     * DOI
-     * is
-     * short
-     * DOI
-     * or
-     * not
-     *
-     * @return true
-     * if
-     * DOI
-     * is
-     * short
-     * DOI,
-     * false
-     * otherwise
+     * Determines whether DOI is short DOI or not
+     * @return true if DOI is short DOI, false otherwise
      */
     public boolean isShortDoi() {
         return isShortDoi;
     }
 
     /**
-     * Return
-     * a
-     * URI
-     * presentation
-     * for
-     * the
-     * DOI/Short
-     * DOI
-     *
-     * @return an
-     * encoded
-     * URI
-     * representation
-     * of
-     * the
-     * DOI/Short
-     * DOI
+     * Return a URI presentation for the DOI/Short DOI
+     * @return an encoded URI representation of the DOI/Short DOI
      */
     @Override
     public Optional<URI> getExternalURI() {
@@ -410,8 +252,7 @@ public class DOI implements Identifier {
         try {
             URI uri = new URI(base.getScheme(), base.getHost(), "/" + doi, null);
             return Optional.of(uri);
-        } catch (
-                URISyntaxException e) {
+        } catch (URISyntaxException e) {
             // should never happen
             LOGGER.error(doi + " could not be encoded as URI.", e);
             return Optional.empty();
@@ -419,24 +260,8 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Return
-     * an
-     * ASCII
-     * URL
-     * presentation
-     * for
-     * the
-     * DOI/Short
-     * DOI
-     *
-     * @return an
-     * encoded
-     * URL
-     * representation
-     * of
-     * the
-     * DOI/Short
-     * DOI
+     * Return an ASCII URL presentation for the DOI/Short DOI
+     * @return an encoded URL representation of the DOI/Short DOI
      */
     public String getURIAsASCIIString() {
         return getExternalURI().map(URI::toASCIIString).orElse("");
@@ -453,13 +278,7 @@ public class DOI implements Identifier {
     }
 
     /**
-     * DOIs
-     * are
-     * case-insensitive.
-     * Thus,
-     * 10.1109/cloud.2017.89
-     * equals
-     * 10.1109/CLOUD.2017.89
+     * DOIs are case-insensitive. Thus, 10.1109/cloud.2017.89 equals 10.1109/CLOUD.2017.89
      */
     @Override
     public boolean equals(Object o) {

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -105,7 +105,7 @@ public class DOI implements Identifier {
     private boolean isShortDoi = false;
 
     /**
-     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs
+     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs.
      *
      * @param doi the DOI/Short DOI string
      * @throws NullPointerException     if DOI/Short DOI is null

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -168,6 +168,10 @@ public class DOI implements Identifier {
             cleanedDOI = formatter.format(cleanedDOI);
             cleanedDOI = cleanedDOI.replaceAll(CHARS_TO_REMOVE, "");
 
+            if (cleanedDOI.startsWith("_") && cleanedDOI.endsWith("_")) {
+                cleanedDOI = cleanedDOI.substring(1, cleanedDOI.length() - 1);
+            }
+
             return Optional.of(new DOI(cleanedDOI));
         } catch (IllegalArgumentException | NullPointerException e) {
             return Optional.empty();

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -10,6 +10,7 @@ import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.jabref.architecture.AllowedToUseLogic;
 import org.jabref.logic.layout.format.LatexToUnicodeFormatter;
 import org.jabref.model.entry.field.Field;
 import org.jabref.model.entry.field.StandardField;
@@ -20,6 +21,7 @@ import org.slf4j.LoggerFactory;
 /**
  * Class for working with <a href="https://en.wikipedia.org/wiki/Digital_object_identifier">Digital object identifiers (DOIs)</a> and <a href="http://shortdoi.org">Short DOIs</a>
  */
+@AllowedToUseLogic("because we want to have this class 'smart' an be able to parse obscure DOIs, too. For this, we need the LatexToUnicodeformatter.")
 public class DOI implements Identifier {
 
     public static final URI AGENCY_RESOLVER = URI.create("https://doi.org/doiRA");

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -2,12 +2,15 @@ package org.jabref.model.entry.identifier;
 
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
 import java.util.Locale;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.jabref.logic.layout.format.LatexToUnicodeFormatter;
 import org.jabref.model.entry.field.Field;
 import org.jabref.model.entry.field.StandardField;
 
@@ -15,7 +18,19 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Class for working with <a href="https://en.wikipedia.org/wiki/Digital_object_identifier">Digital object identifiers (DOIs)</a> and <a href="http://shortdoi.org">Short DOIs</a>
+ * Class
+ * for
+ * working
+ * with
+ * <a
+ * href="https://en.wikipedia.org/wiki/Digital_object_identifier">Digital
+ * object
+ * identifiers
+ * (DOIs)</a>
+ * and
+ * <a
+ * href="http://shortdoi.org">Short
+ * DOIs</a>
  */
 public class DOI implements Identifier {
 
@@ -93,8 +108,8 @@ public class DOI implements Identifier {
     // See https://stackoverflow.com/questions/3203190/regex-any-ascii-character for the regexp that includes ASCII characters only
     // Another reference for regexp for ASCII characters: https://howtodoinjava.com/java/regex/java-clean-ascii-text-non-printable-chars/
     private static final String CHARS_TO_REMOVE = "[\\s+" // remove white space characters, i.e, \t, \n, \x0B, \f, \r . + is a greedy quantifier
-                                                + "[^\\x00-\\x7F]" // strips off all non-ASCII characters
-                                                + "]";
+            + "[^\\x00-\\x7F]" // strips off all non-ASCII characters
+            + "]";
 
     // DOI
     private final String doi;
@@ -102,11 +117,38 @@ public class DOI implements Identifier {
     private boolean isShortDoi = false;
 
     /**
-     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs.
+     * Creates
+     * a
+     * DOI
+     * from
+     * various
+     * schemes
+     * including
+     * URL,
+     * URN,
+     * and
+     * plain
+     * DOIs/Short
+     * DOIs.
      *
-     * @param doi the DOI/Short DOI string
-     * @throws NullPointerException     if DOI/Short DOI is null
-     * @throws IllegalArgumentException if doi does not include a valid DOI/Short DOI
+     * @param doi the
+     *            DOI/Short
+     *            DOI
+     *            string
+     * @throws NullPointerException     if
+     *                                  DOI/Short
+     *                                  DOI
+     *                                  is
+     *                                  null
+     * @throws IllegalArgumentException if
+     *                                  doi
+     *                                  does
+     *                                  not
+     *                                  include
+     *                                  a
+     *                                  valid
+     *                                  DOI/Short
+     *                                  DOI
      */
     public DOI(String doi) {
         Objects.requireNonNull(doi);
@@ -120,7 +162,8 @@ public class DOI implements Identifier {
                 // decodes path segment
                 URI url = new URI(trimmedDoi);
                 trimmedDoi = url.getScheme() + "://" + url.getHost() + url.getPath();
-            } catch (URISyntaxException e) {
+            } catch (
+                    URISyntaxException e) {
                 throw new IllegalArgumentException(doi + " is not a valid HTTP DOI/Short DOI.");
             }
         }
@@ -150,39 +193,119 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Creates an Optional&lt;DOI> from various schemes including URL, URN, and plain DOIs.
+     * Creates
+     * an
+     * Optional&lt;DOI>
+     * from
+     * various
+     * schemes
+     * including
+     * URL,
+     * URN,
+     * and
+     * plain
+     * DOIs.
      * <p>
-     * Useful for suppressing the <c>IllegalArgumentException</c> of the Constructor and checking for Optional.isPresent() instead.
+     * Useful
+     * for
+     * suppressing
+     * the
+     * <c>IllegalArgumentException</c>
+     * of
+     * the
+     * Constructor
+     * and
+     * checking
+     * for
+     * Optional.isPresent()
+     * instead.
      *
-     * @param doi the DOI/Short DOI string
-     * @return an Optional containing the DOI or an empty Optional
+     * @param doi the
+     *            DOI/Short
+     *            DOI
+     *            string
+     * @return an
+     * Optional
+     * containing
+     * the
+     * DOI
+     * or
+     * an
+     * empty
+     * Optional
      */
     public static Optional<DOI> parse(String doi) {
         try {
+            LatexToUnicodeFormatter formatter = new LatexToUnicodeFormatter();
             String cleanedDOI = doi;
+            cleanedDOI = URLDecoder.decode(cleanedDOI, StandardCharsets.UTF_8);
+            cleanedDOI = formatter.format(cleanedDOI);
             cleanedDOI = cleanedDOI.replaceAll(CHARS_TO_REMOVE, "");
 
             return Optional.of(new DOI(cleanedDOI));
-        } catch (IllegalArgumentException | NullPointerException e) {
+        } catch (
+                IllegalArgumentException |
+                NullPointerException e) {
             return Optional.empty();
         }
     }
 
     /**
-     * Determines whether a DOI/Short DOI is valid or not
+     * Determines
+     * whether
+     * a
+     * DOI/Short
+     * DOI
+     * is
+     * valid
+     * or
+     * not
      *
-     * @param doi the DOI/Short DOI string
-     * @return true if DOI is valid, false otherwise
+     * @param doi the
+     *            DOI/Short
+     *            DOI
+     *            string
+     * @return true
+     * if
+     * DOI
+     * is
+     * valid,
+     * false
+     * otherwise
      */
     public static boolean isValid(String doi) {
         return parse(doi).isPresent();
     }
 
     /**
-     * Tries to find a DOI/Short DOI inside the given text.
+     * Tries
+     * to
+     * find
+     * a
+     * DOI/Short
+     * DOI
+     * inside
+     * the
+     * given
+     * text.
      *
-     * @param text the Text which might contain a DOI/Short DOI
-     * @return an Optional containing the DOI or an empty Optional
+     * @param text the
+     *             Text
+     *             which
+     *             might
+     *             contain
+     *             a
+     *             DOI/Short
+     *             DOI
+     * @return an
+     * Optional
+     * containing
+     * the
+     * DOI
+     * or
+     * an
+     * empty
+     * Optional
      */
     public static Optional<DOI> findInText(String text) {
         Optional<DOI> result = Optional.empty();
@@ -214,27 +337,63 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Return the plain DOI/Short DOI
+     * Return
+     * the
+     * plain
+     * DOI/Short
+     * DOI
      *
-     * @return the plain DOI/Short DOI value.
+     * @return the
+     * plain
+     * DOI/Short
+     * DOI
+     * value.
      */
     public String getDOI() {
         return doi;
     }
 
     /**
-     * Determines whether DOI is short DOI or not
+     * Determines
+     * whether
+     * DOI
+     * is
+     * short
+     * DOI
+     * or
+     * not
      *
-     * @return true if DOI is short DOI, false otherwise
+     * @return true
+     * if
+     * DOI
+     * is
+     * short
+     * DOI,
+     * false
+     * otherwise
      */
     public boolean isShortDoi() {
         return isShortDoi;
     }
 
     /**
-     * Return a URI presentation for the DOI/Short DOI
+     * Return
+     * a
+     * URI
+     * presentation
+     * for
+     * the
+     * DOI/Short
+     * DOI
      *
-     * @return an encoded URI representation of the DOI/Short DOI
+     * @return an
+     * encoded
+     * URI
+     * representation
+     * of
+     * the
+     * DOI/Short
+     * DOI
      */
     @Override
     public Optional<URI> getExternalURI() {
@@ -251,7 +410,8 @@ public class DOI implements Identifier {
         try {
             URI uri = new URI(base.getScheme(), base.getHost(), "/" + doi, null);
             return Optional.of(uri);
-        } catch (URISyntaxException e) {
+        } catch (
+                URISyntaxException e) {
             // should never happen
             LOGGER.error(doi + " could not be encoded as URI.", e);
             return Optional.empty();
@@ -259,9 +419,24 @@ public class DOI implements Identifier {
     }
 
     /**
-     * Return an ASCII URL presentation for the DOI/Short DOI
+     * Return
+     * an
+     * ASCII
+     * URL
+     * presentation
+     * for
+     * the
+     * DOI/Short
+     * DOI
      *
-     * @return an encoded URL representation of the DOI/Short DOI
+     * @return an
+     * encoded
+     * URL
+     * representation
+     * of
+     * the
+     * DOI/Short
+     * DOI
      */
     public String getURIAsASCIIString() {
         return getExternalURI().map(URI::toASCIIString).orElse("");
@@ -278,7 +453,13 @@ public class DOI implements Identifier {
     }
 
     /**
-     * DOIs are case-insensitive. Thus, 10.1109/cloud.2017.89 equals 10.1109/CLOUD.2017.89
+     * DOIs
+     * are
+     * case-insensitive.
+     * Thus,
+     * 10.1109/cloud.2017.89
+     * equals
+     * 10.1109/CLOUD.2017.89
      */
     @Override
     public boolean equals(Object o) {

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -105,7 +105,8 @@ public class DOI implements Identifier {
     private boolean isShortDoi = false;
 
     /**
-     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs.
+     * Creates a DOI from various schemes including URL, URN, and plain DOIs/Short DOIs
+     *
      * @param doi the DOI/Short DOI string
      * @throws NullPointerException     if DOI/Short DOI is null
      * @throws IllegalArgumentException if doi does not include a valid DOI/Short DOI
@@ -154,7 +155,9 @@ public class DOI implements Identifier {
 
     /**
      * Creates an Optional&lt;DOI> from various schemes including URL, URN, and plain DOIs.
+     * <p>
      * Useful for suppressing the <c>IllegalArgumentException</c> of the Constructor and checking for Optional.isPresent() instead.
+     *
      * @param doi the DOI/Short DOI string
      * @return an Optional containing the DOI or an empty Optional
      */
@@ -176,6 +179,7 @@ public class DOI implements Identifier {
 
     /**
      * Determines whether a DOI/Short DOI is valid or not
+     *
      * @param doi the DOI/Short DOI string
      * @return true if DOI is valid, false otherwise
      */
@@ -185,6 +189,7 @@ public class DOI implements Identifier {
 
     /**
      * Tries to find a DOI/Short DOI inside the given text.
+     *
      * @param text the Text which might contain a DOI/Short DOI
      * @return an Optional containing the DOI or an empty Optional
      */
@@ -219,6 +224,7 @@ public class DOI implements Identifier {
 
     /**
      * Return the plain DOI/Short DOI
+     *
      * @return the plain DOI/Short DOI value.
      */
     public String getDOI() {
@@ -227,6 +233,7 @@ public class DOI implements Identifier {
 
     /**
      * Determines whether DOI is short DOI or not
+     *
      * @return true if DOI is short DOI, false otherwise
      */
     public boolean isShortDoi() {
@@ -235,6 +242,7 @@ public class DOI implements Identifier {
 
     /**
      * Return a URI presentation for the DOI/Short DOI
+     *
      * @return an encoded URI representation of the DOI/Short DOI
      */
     @Override
@@ -261,6 +269,7 @@ public class DOI implements Identifier {
 
     /**
      * Return an ASCII URL presentation for the DOI/Short DOI
+     *
      * @return an encoded URL representation of the DOI/Short DOI
      */
     public String getURIAsASCIIString() {

--- a/src/main/java/org/jabref/model/entry/identifier/DOI.java
+++ b/src/main/java/org/jabref/model/entry/identifier/DOI.java
@@ -123,8 +123,7 @@ public class DOI implements Identifier {
                 // decodes path segment
                 URI url = new URI(trimmedDoi);
                 trimmedDoi = url.getScheme() + "://" + url.getHost() + url.getPath();
-            } catch (
-                    URISyntaxException e) {
+            } catch (URISyntaxException e) {
                 throw new IllegalArgumentException(doi + " is not a valid HTTP DOI/Short DOI.");
             }
         }
@@ -170,9 +169,7 @@ public class DOI implements Identifier {
             cleanedDOI = cleanedDOI.replaceAll(CHARS_TO_REMOVE, "");
 
             return Optional.of(new DOI(cleanedDOI));
-        } catch (
-                IllegalArgumentException |
-                NullPointerException e) {
+        } catch (IllegalArgumentException | NullPointerException e) {
             return Optional.empty();
         }
     }

--- a/src/test/java/org/jabref/logic/cleanup/DoiCleanupTest.java
+++ b/src/test/java/org/jabref/logic/cleanup/DoiCleanupTest.java
@@ -66,7 +66,6 @@ public class DoiCleanupTest {
                 Arguments.of(new BibEntry()
                          .withField(StandardField.DOI, "10.18726/2018_3"),
                           new BibEntry()
-                         .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3"))
-        );
-    }
+                         .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")));
+            }
 }

--- a/src/test/java/org/jabref/logic/cleanup/DoiCleanupTest.java
+++ b/src/test/java/org/jabref/logic/cleanup/DoiCleanupTest.java
@@ -60,7 +60,13 @@ public class DoiCleanupTest {
 
                 // cleanup just ee field with URL
                 Arguments.of(doiResult, new BibEntry()
-                        .withField(unknownField, "https://doi.org/10.1145/2594455"))
+                        .withField(unknownField, "https://doi.org/10.1145/2594455")),
+
+                // cleanup of url encoded chars
+                Arguments.of(new BibEntry()
+                         .withField(StandardField.DOI, "10.18726/2018_3"),
+                          new BibEntry()
+                         .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3"))
         );
     }
 }

--- a/src/test/java/org/jabref/logic/cleanup/DoiDecodeCleanupTest.java
+++ b/src/test/java/org/jabref/logic/cleanup/DoiDecodeCleanupTest.java
@@ -1,0 +1,67 @@
+package org.jabref.logic.cleanup;
+
+import java.util.stream.Stream;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.field.UnknownField;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DoiDecodeCleanupTest {
+
+    @ParameterizedTest
+    @MethodSource("provideDoiForAllLowers")
+    public void testChangeDoi(BibEntry expected, BibEntry doiInfoField) {
+        DoiCleanup cleanUp = new DoiCleanup();
+        cleanUp.cleanup(doiInfoField);
+
+        assertEquals(expected, doiInfoField);
+    }
+
+    private static Stream<Arguments> provideDoiForAllLowers() {
+        UnknownField unknownField = new UnknownField("ee");
+        BibEntry doiResult = new BibEntry().withField(StandardField.DOI, "10.18726/2018_3");
+
+        return Stream.of(
+                // cleanup for Doi field only
+                Arguments.of(doiResult, new BibEntry().withField(
+                        StandardField.URL, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")),
+
+                // cleanup with Doi and URL to all entries
+                Arguments.of(doiResult, new BibEntry()
+                        .withField(StandardField.DOI, "10.18726/2018%7B%5Ctextunderscore%7D3")
+                        .withField(StandardField.URL, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")
+                        .withField(StandardField.NOTE, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")
+                        .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")),
+
+                // cleanup with Doi and no URL to entries
+                Arguments.of(
+                        new BibEntry()
+                        .withField(StandardField.DOI, "10.18726/2018_3")
+                        .withField(StandardField.NOTE, "This is a random note to this Doi")
+                        .withField(unknownField, "This is a random ee field for this Doi"),
+                        new BibEntry()
+                        .withField(StandardField.DOI, "10.18726/2018_3")
+                        .withField(StandardField.NOTE, "This is a random note to this Doi")
+                        .withField(unknownField, "This is a random ee field for this Doi")),
+
+                // cleanup with spaced Doi
+                Arguments.of(doiResult, new BibEntry()
+                        .withField(StandardField.DOI, "10.18726/2018%7B%5Ctextunderscore%7D3")),
+
+                // cleanup just Note field with URL
+                Arguments.of(doiResult, new BibEntry()
+                        .withField(StandardField.NOTE, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3")),
+
+                // cleanup just ee field with URL
+                Arguments.of(doiResult, new BibEntry()
+                        .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3"))
+        );
+    }
+
+}

--- a/src/test/java/org/jabref/logic/cleanup/DoiDecodeCleanupTest.java
+++ b/src/test/java/org/jabref/logic/cleanup/DoiDecodeCleanupTest.java
@@ -63,5 +63,4 @@ public class DoiDecodeCleanupTest {
                         .withField(unknownField, "https://doi.org/10.18726/2018%7B%5Ctextunderscore%7D3"))
         );
     }
-
 }

